src/workloads/execution/FilterWithComplexLogicalExpression.yml

```diff
@@ -32,7 +32,34 @@ GlobalDefaults:
   DocumentCount: &DocumentCount 1e6
   Repeat: &Repeat 50
   Threads: &Threads 1
-  MaxPhases: &MaxPhases 17
+  MaxPhases: &MaxPhases 23
```

Comment:
I actually suspect that this can be 22, not 23. I doubt it matters though.

Also, it's somewhat unfortunate that we need to keep maintaining this number... We don't necessarily have to do it as part of this patch, but I wonder if there is a way to write this workload as a single phase. The phase would be a single `CrudActor` that runs multiple find operations one after the other and measures each independently. I think this kind of thing is supported by the `CrudActor`, if I understand correctly. Theoretically this would get rid of some of the boilerplate in this patch as well.

Reply:
> Yeah, I'm not sure if if this is 0 indexed or not (i.e. is it the number of phases, or is it the highest phase number that we reach?)

I looked into this when we had a similar discussion on my genny patch for the $group spilling workloads. I'm 95% certain that it's the highest phase number we reach (zero-indexed) as opposed to the number of phases (one-indexed). That appears consistent with the latest code in this patch, so I think we can consider this topic resolved.

> Yeah; we could totally refactor these. I think it would look something like this. I can file a ticket and stick it in m4 if you would like.

I don't think it's worth messing with this particular workload, because it already has a different structure where each query has its own `CrudActor` declaration. So I wouldn't bother filing a ticket for m4. I doubt we'd actually do it. Going forward, however, one could argue that it would be better to write such workloads using a single `CrudActor` that executes multiple queries in sequence.

Reply:
Feel free to resolve this thread after reading my previous response.